agent/consul/server_test.go

```diff
@@ -2100,3 +2105,39 @@ func TestServer_hcpManager(t *testing.T) {
 	hcp1.AssertExpectations(t)
 
 }
+
+// goldenMarkdown reads and optionally writes the expected data to the goldenMarkdown file,
+// returning the contents as a string.
+func goldenMarkdown(t *testing.T, name, got string) string {
+	t.Helper()
+
+	golden := filepath.Join("testdata", name+".md")
+	update := flag.Lookup("update").Value.(flag.Getter).Get().(bool)
+	if update && got != "" {
+		err := os.WriteFile(golden, []byte(got), 0644)
+		require.NoError(t, err)
+	}
+
+	expected, err := os.ReadFile(golden)
+	require.NoError(t, err)
+
+	return string(expected)
+}
+
+func TestServer_ControllerDependencies(t *testing.T) {
+	t.Parallel()
+
+	_, conf := testServerConfig(t)
+	deps := newDefaultDeps(t, conf)
+	deps.Experiments = []string{"resource-apis"}
+	deps.LeafCertManager = &leafcert.Manager{}
+
+	s1, err := newServerWithDeps(t, conf, deps)
+	require.NoError(t, err)
+	defer s1.Shutdown()
```

Comment:
nit: the defer shutdown hasn't bee needed for awhile now that we use `t.Cleanup` internally to do that.